### PR TITLE
Remove RevenueCat.com

### PR DESCRIPTION
**Pull request by francocorreasosa**
Hey! 👋🏻 I work for RevenueCat.com and we just noticed that we were added to this list.

RevenueCat is an in-app purchase processing system and is not used to serve ads to users (actually, some apps use us to let users remove ads!) Blocking our API prevents people from being able to make purchases and unlock great content and features on thousands of apps.

It would be great if we could remove RevenueCat from this list, since lots of users would be getting errors and asking for support from both RC and our customers, with us being unable to help. Actually, we have been getting some reports already that some of our customers' apps are broken because of this.

Let us know your thoughts, I’m happy to answer any questions about our service if you need it.

Thanks!

### CODE REVIEW

**Comment by dnmTX** [comment]
**ping** @jawz101 for assist and evaluation. Thanks 👍

**Comment by jawz101** [comment]
According to the FAQ's you are not a payment processor

https://www.revenuecat.com/faq

According to the revenuecat site, it is used to segment users based on purchases and demographic identifiers to influence purchasing decisions.

Examples in the documentation include letting apps collect age, name, phone number, favorite food, email, hair color, and what sounds like anything an app wants to collect.

https://docs.revenuecat.com/docs/subscriber-attributes


These blocklists aren't just about blocking an ad on a webpage.  The more pernicious problem is there is no universal law that let me prevent collection of, notify of what has been collected, as well as request my data be removed from company A, their 3rd parties, nor the 3rd parties of their 3rd parties.  The next thing I know, my name, favorite food, hair color, and whatever is known by 1,000 companies because one company thought it was OK for them to pass it around to make money off of that info for the sake of raising a price on me, upselling me stuff I don't want, keeping my attention on something I don't want.


How about focusing on contextual marketing instead of behavioral marketing?  That is, instead of knowing that my identity (including characteristics and behaviors) is using a product- focus on the product and its relevant, related thingies.  50 years ago if you picked up a Men's magazine the ads were for car parts, cologne, workout supplements, and such because it was a magazine for dudes.  Now, you look at that Men's magazine and it shows you a tickler ad because you eyed a toy for your cat or had a USB charger cable in your Amazon shopping cart.  And 100 companies know it as well.

**Comment by dnmTX** [comment]
@jawz101 thanks for this ☝️ ❤️ !!!! 
And one more thing(it's off topic so apologies Steve),i saw that invitation too late(sorry about that),so if you want to resend it i'll make sure to activate it ASAP 👍

**Comment by jeiting** [comment]
We do provide some general purpose data storage facilities for apps that want to use them, but none of that collection is done by us automatically. Do you block all general purpose databases? Do you block every app's backend? Just curious where the line is. Do you block Firebase and Supabase?

As for us "not being a payment processor." Without being pedantic, blocking RevenueCat will be cutting off the revenue stream. Our service hosts the SKUs, facilitates the payment, verifies the purchase, and unlocks entitlement. If preventing a commercial relationship with apps is your goal, then so be it. We are just trying to help our developers make money here, so they don't have to do things like sell your data.

**Comment by jawz101** [comment]
api.revenuecat.com looks to be the [6,926 most requested DNS lookup](http://s3-us-west-1.amazonaws.com/umbrella-static/index.html) of the 100's of millions of users who use Cisco Umbrella/OpenDNS resolvers in the world.  Congratulations.  That's not the top 6,926 websites- it's DNS lookups.  That's like more than people search for cruises on Carnival Cruise, plan for trips on Trip Advisor, download updates from Microsoft, visit Hulu, Vudu, American Express, Fox Business, Verizon WIreless... you get the idea.

These lists are nothing and users are well aware they can allow something that they want to allow.

**Comment by StevenBlack** [comment]
Well, wow. Thanks for this @jawz101.

So Jacob @jeiting here's the deal. We serve users who don't want to be traded or tracked and, of course, assailed by ads, phished, and worse. It's really that simple.

I understand what you guys do.  I'm sure that's legit, and I understand why.

You're just on the wrong side of the line.  We make decisions like this all the time. I hope you understand where we're coming from. People depend on us, and one of the things we do is, draw the line.

Closing.

**Comment by jeiting** [comment]
Steven, I appreciate your respectful tone over being talked down to because our service doesn’t outrank Carnival Cruises.

It’s fine if you have a line, and it’s fine if you hold it. I’d just appreciate some consistency. Developers use our infrastructure to make money in a very honest, direct, and privacy conscious way. Our SDK is designed by default to be collect as little information as possible to do the job.

If this is your line, that any service that *can* collect IID, belongs on your list, I’d ask that you apply it equally, and block Firebase as well. It’s Google after all and they are blatant about their harvesting of data for profit. 

Anyway, this is the last time I’ll ask you to reconsider. Thanks for your time.

**Comment by StevenBlack** [comment]
Thanks Jacob @jeiting.

But... we block lots of Firebase, and a ton of Google.

```
$ ack firebase

hosts
5547:0.0.0.0 firebase-settings.crashlytics.com
6697:# [firebaseio.com]
6698:0.0.0.0 famobi-01.firebaseio.com
36090:0.0.0.0 stockbook-ads.firebaseapp.com
36091:0.0.0.0 stockbook-ads.firebaseio.com
40614:0.0.0.0 akpuruachi.firebaseapp.com
47263:0.0.0.0 iyeuryetts.firebaseapp.com
47539:0.0.0.0 kg-hm-p.firebaseapp.com
48195:0.0.0.0 lot-as.firebaseapp.com
48201:0.0.0.0 lot-pl-x.firebaseapp.com
48700:0.0.0.0 mirror-lp.firebaseapp.com
50503:0.0.0.0 orlen-p.firebaseapp.com
50515:0.0.0.0 orlen-q.firebaseapp.com
53384:0.0.0.0 signifyteleprompt-expired.firebaseapp.com
54275:0.0.0.0 tesla-x-pl.firebaseapp.com
55802:0.0.0.0 www.akpuruachi.firebaseapp.com
60896:0.0.0.0 www.iyeuryetts.firebaseapp.com
68950:0.0.0.0 www.kg-hm-p.firebaseapp.com
69133:0.0.0.0 www.lot-as.firebaseapp.com
69139:0.0.0.0 www.lot-pl-x.firebaseapp.com
69246:0.0.0.0 www.mirror-lp.firebaseapp.com
69862:0.0.0.0 www.orlen-p.firebaseapp.com
69874:0.0.0.0 www.orlen-q.firebaseapp.com
70900:0.0.0.0 www.signifyteleprompt-expired.firebaseapp.com
71097:0.0.0.0 www.tesla-x-pl.firebaseapp.com
86232:0.0.0.0 firebaselogging.googleapis.com
86402:0.0.0.0 mononoteapp.firebaseio.com
91488:0.0.0.0 ads-game-187f4.firebaseapp.com
```


```
$ ack google

hosts
223:0.0.0.0 auth-google.site
343:0.0.0.0 adservice.google.co.in
360:0.0.0.0 adservice.google.com.vn
864:0.0.0.0 fgoogle.com
865:0.0.0.0 www.fgoogle.com
917:0.0.0.0 adservice.google.be
918:0.0.0.0 marketingplatform.google.com
919:0.0.0.0 smartlock.google.com
920:0.0.0.0 bot2.googlerenewals.net
921:0.0.0.0 linux2.googlerenewals.net
922:0.0.0.0 www.googletagmanager.com
923:0.0.0.0 cdn.googletoolservices.com
1877:0.0.0.0 ssl.googleanalytics.com
1878:0.0.0.0 click.googleanalytics.com
1879:0.0.0.0 alt2.googleanalytics.com
1880:0.0.0.0 alt4.googleanalytics.com
1881:0.0.0.0 sandbox.googleanalytics.com
6081:0.0.0.0 googleads.g.doubleclick.net
6082:0.0.0.0 googleads4.g.doubleclick.net
6938:0.0.0.0 googlecm.hit.gemius.pl
7087:# [google-analytics.com]
7088:0.0.0.0 google-analytics.com
7089:0.0.0.0 ssl.google-analytics.com
7090:0.0.0.0 www.google-analytics.com
7092:# [google.com]
7093:0.0.0.0 ads.google.com
7094:0.0.0.0 adservice.google.com
7095:0.0.0.0 crashlytics2.l.google.com
7096:0.0.0.0 s0-2mdn-net.l.google.com
7098:# [googleadservices.com]
7099:0.0.0.0 googleadservices.com
7100:0.0.0.0 pagead2.googleadservices.com
7101:0.0.0.0 www.googleadservices.com
7103:# [googleadsserving.cn]
7104:0.0.0.0 static.googleadsserving.cn
7106:# [crashlyticsreports-pa.googleapis.com]
7107:0.0.0.0 crashlyticsreports-pa.googleapis.com
7109:# [googlesyndication.com]
7110:0.0.0.0 googlesyndication.com
7111:0.0.0.0 ade.googlesyndication.com
7112:0.0.0.0 pagead2.googlesyndication.com
7113:0.0.0.0 tpc.googlesyndication.com
7115:# [googleusercontent.com]
7116:0.0.0.0 displayads-formats.googleusercontent.com
15893:0.0.0.0 google-docs.org
16714:0.0.0.0 www-google.nl
17006:0.0.0.0 www.google-docs.org
17007:0.0.0.0 www.googlew.com
17417:0.0.0.0 www.www-google.nl
17430:0.0.0.0 www.wwwgoogles.com
17479:0.0.0.0 wwwgoogles.com
26185:# newssuite-sony-net.storage.googleapis.com disguised trackers
27202:0.0.0.0 googleads.g.doubleclick.net.iberostar.com
33313:0.0.0.0 13453765871837679316.googlegroups.com
35793:0.0.0.0 google-shopping.sapoapps.vn
35794:0.0.0.0 googleshopping.sapoapps.vn
45980:0.0.0.0 googleailesi.com
45981:0.0.0.0 googleinfo.coocoldini.xyz
45982:0.0.0.0 google-deckoffice.ml
45983:0.0.0.0 google-yandex.info
46698:0.0.0.0 info-google.waw.pl
51395:0.0.0.0 plgoogle.lvlgoodnewspl.site
53835:0.0.0.0 store.apps.googleplay.pro
55379:0.0.0.0 wiadomosci-google.pl
55380:0.0.0.0 wiadomosci-google.waw.pl
60133:0.0.0.0 www.googleailesi.com
60134:0.0.0.0 www.googleinfo.coocoldini.xyz
63676:0.0.0.0 www.plgoogle.lvlgoodnewspl.site
65303:0.0.0.0 www.store.apps.googleplay.pro
68373:0.0.0.0 www.google-deckoffice.ml
68374:0.0.0.0 www.google-yandex.info
68530:0.0.0.0 www.info-google.waw.pl
71363:0.0.0.0 www.wiadomosci-google.pl
71364:0.0.0.0 www.wiadomosci-google.waw.pl
71458:0.0.0.0 www.wydarzenia-google.waw.pl
71620:0.0.0.0 wydarzenia-google.waw.pl
77676:0.0.0.0 mt34.mtree.com  #[Google.Diagnostic]
78457:# 0.0.0.0 wurfl.io #[affects Google]
79900:# [Google][AS15169][104.154.0.0 - 104.199.255.255]
79912:# [Google][AS15169][107.178.192.0 - 107.178.255.255]
79915:# [Google][AS15169][108.170.192.0 - 108.170.255.255]
79917:# [Google][AS15169][130.211.0.0 - 130.211.255.255]
79933:# [Google][AS15169][172.217.0.0 - 172.217.255.255]
79946:0.0.0.0 pagead-googlehosted.l.google.com
79947:# [Google][AS15169][173.194.0.0 - 173.194.255.255]
79949:0.0.0.0 video-ad-stats.googlesyndication.com
79950:# 0.0.0.0 redirector.googlevideo.com #[affects video playback]
79951:# [Google / DoubleClick][AS15169][173.194.0.0 - 173.194.255.255]
79954:# [Google / DoubleClick][AS15169][209.85.128.0 - 209.85.255.255]
79955:0.0.0.0 analytics-api-samples.googlecode.com
79956:# [Google][AS15169][216.58.192.0 - 216.58.223.255]
79962:# 0.0.0.0 www.googletagmanager.com #[affects ASUS]
79963:# [Google / DoubleClick][AS6432][216.73.80.0 - 216.73.95.255]
79965:# [Google][AS15169][23.236.128.0 - 23.251.159.255]
79966:# [Google][AS15169][34.64.0.0/11]
79969:# [Google][AS15169][35.184.0.0 - 35.240.255.255]
79978:# [Google / DoubleClick][AS15169][64.233.160.0 - 64.233.191.255]
80049:# [Google / DoubleClick][AS15169][66.249.64.0 - 66.249.95.255]
80051:# [Google / DoubleClick][AS15169][72.14.192.0 - 72.14.255.255]
80059:0.0.0.0 googleads2.g.doubleclick.net
80061:# [Google / DoubleClick][AS15169][74.125.0.0 - 74.125.255.255]
80068:0.0.0.0 www.doubleclickbygoogle.com
80069:0.0.0.0 video-stats.video.google.com
80070:0.0.0.0 4.afs.googleadservices.com
80071:0.0.0.0 partner.googleadservices.com
80072:0.0.0.0 www.googletagservices.com
80888:0.0.0.0 blueadvertise.com  #[Google.Diagnostic][2013-01-02]
85451:# killedbygoogle.com
85741:#74.125.127.105	g        # google.com
85785:#0.0.0.0 google-analytics.com			# breaks some sites
85786:#0.0.0.0 ssl.google-analytics.com
85787:#0.0.0.0 www.google-analytics.l.google.com
85792:#0.0.0.0 www.google-analytics.com		# breaks some sites
85806:#0.0.0.0 www.googletagservices.com		#interferes with techrepublic
86232:0.0.0.0 firebaselogging.googleapis.com
86261:0.0.0.0 googletagservices.com
87273:0.0.0.0 adwords.google.lloymlincs.com
88184:0.0.0.0 www.adwords.google.lloymlincs.com
89753:0.0.0.0 googleplay-store.com
92685:0.0.0.0 adservice.google.ca
92686:0.0.0.0 adservice.google.co.za
92687:0.0.0.0 adservice.google.com.au
92688:0.0.0.0 adservice.google.cz
92689:0.0.0.0 adservice.google.nl
92690:0.0.0.0 adservices.google.com
92885:0.0.0.0 affiliate.googleusercontent.com
93438:0.0.0.0 buttons.googlesyndication.com
94137:0.0.0.0 drm-google-analtyic.com
94333:0.0.0.0 feedads.googleadservices.com
94676:0.0.0.0 googledrive-download.com
94677:0.0.0.0 googledrive-en.com
94678:0.0.0.0 googledrive-eu.com
94679:0.0.0.0 googledrive-gb.com
94680:0.0.0.0 googlewordpad.info
95393:0.0.0.0 m4.afs.googleadservices.com
95593:0.0.0.0 mimageads.googleadservices.com
95594:0.0.0.0 mimageads1.googleadservices.com
95595:0.0.0.0 mimageads2.googleadservices.com
95596:0.0.0.0 mimageads3.googleadservices.com
95597:0.0.0.0 mimageads4.googleadservices.com
95598:0.0.0.0 mimageads5.googleadservices.com
95599:0.0.0.0 mimageads6.googleadservices.com
95600:0.0.0.0 mimageads7.googleadservices.com
95601:0.0.0.0 mimageads8.googleadservices.com
95602:0.0.0.0 mimageads9.googleadservices.com
95671:0.0.0.0 mpartner.googleadservices.com
95824:0.0.0.0 news-server-drm-google.com
96107:0.0.0.0 pagead.googlesyndication.com
96108:0.0.0.0 pagead1.googlesyndication.com
96109:0.0.0.0 pagead3.googlesyndication.com
97116:0.0.0.0 static-google-analtyic.com
98493:0.0.0.0 www.googlewordpad.info
98747:0.0.0.0 www.partner.googleadservices.com
100685:0.0.0.0 router.googlecom.biz
101213:0.0.0.0 adservice.google.co.jp
102165:0.0.0.0 ssl-google-analytics.l.google.com
102751:0.0.0.0 drive.google.com.it-barcelona.com
104229:0.0.0.0 adservice.google.com.mt
104377:0.0.0.0 analytics.google.com
104668:0.0.0.0 clientmetrics-pa.googleapis.com
105042:0.0.0.0 googleanalytics.com
105043:0.0.0.0 googletagmanager.com
105338:0.0.0.0 mail-ads.google.com
105578:0.0.0.0 pagead.l.google.com
105593:0.0.0.0 partnerad.l.google.com
106370:0.0.0.0 wintricksbanner.googlepages.com
106388:0.0.0.0 www-google-analytics.l.google.com
````

**Comment by jeiting** [comment]
You’re missing firebase.googleapis.com 

https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/api/settings.h#L37

Which is the arbitrary key/value storage API that was cited for our inclusion.

Really, I don’t think you should block this since Firebase is a useful tool (that can be used in harmful ways) and it would harm a ton of developers and users if it were blocked. Just pointing it out as a weird arbitrary line.

**Comment by dnmTX** [comment]
**HOLD THE LINE STEVO.....HOOOOOLD THE LINE** 😉 👍

**Comment by rkotzy** [comment]
I understand you need to draw a line somewhere - let me know if there's any info we can provide to help you make an informed decision. I previously worked on a "developer tool" that was an ad-tech data sausage factory on the backend. One of the main reasons I started working on RevenueCat was to build 100% open-source SDKs that only exist to help developers with no ulterior motives, you can go through the code yourself and verify that no data is collected by the system.

Congrats on developing and maintaining such a popular block list - there are a ton of people getting value from this. It does seem irresponsible though to merge PRs without fact checking the claims though since you have the power to hurt consumers as well.

The tracking done by RevenueCat is for in-app purchases that's stored against an anonymous UUID that is not persisted across installs. IP address is dropped from the RevenueCat servers to be COPPA compliant in kids apps. We do not want to track users, it's a big compliance headache and not our business. We track subscription status, which by the design of Apple/Google cannot be linked back to any individual device to preserve privacy on those platforms. All RevenueCat cares about is the subscription status and what anonymous UUID it's tied back to. 

The subscriber attributes referenced in the PR to block api.revenuecat.com are an arbitrary key-value store, no different than [Stripe metadata](https://stripe.com/docs/api/metadata) which does not appear to be blocked. Yes, some developers choose to use this metadata to collect things that would be "across your line" - just like with Stripe - and I can acknowledge that. But it's still not clear to me how that is different than any other database that you don't know how it's being used?

The reports we're getting from developers now is that customers who paid money to access premium features of their app are not getting access. Consumers are getting ripped off and developers trying to earn a living by building quality apps people want to pay for are losing money and getting bad reviews. If someone is getting value from a subscription app, there's probably a chance they disable any blocking tool before they cancel their subscription (sorry) which then opens them up to actual ad/data tracking.

>According to the FAQ's you are not a payment processor

Sure. So is Chargebee, Recurly, etc. that use Stripe under-the-hood.

>According to the revenuecat site, it is used to segment users based on purchases and demographic identifiers to influence purchasing decisions.

The segmentation is done based off of subscription status. (e.g. show me all of my subscriptions that were paid in USD and started between these dates) There's no tools in RevenueCat used to "influence purchasing decisions" and no demographic identifiers associated with a subscription.

>Examples in the documentation include letting apps collect age, name, phone number, favorite food, email, hair color, and what sounds like anything an app wants to collect.

None of this data is provided by RevenueCat, developers need to collect it. Meaning if a fitness app asks for someone's age, **and the customer agrees to give it to them**, the developer can choose to to store it in a key-value store inside of RevenueCat. The key here is all of this information would need to be provided by some other means outside of RevenueCat.

>How about focusing on contextual marketing instead of behavioral marketing? 

RevenueCat does not serve any ads or participate in any of the advertising data exchanges you're referencing here. We focus on empowering developers with free and open-source SDKs to charge for their product instead of having to rely on advertising based models.

>We serve users who don't want to be traded or tracked and, of course, assailed by ads, phished, and worse. It's really that simple.

Amazing! We're all fans of that! Then block the companies that do the tracking/trading/ads. For example Appsflyer: If a developer chose to have a RevenueCat -> Appsflyer integration set up, then by blocking Appsflyer the integration wouldn't work and the consumer is protected. I get what you're trying to do but don't lock out customers who spent money on an app, it sucks for everyone involved.

**Comment by jawz101** [comment]
I walked into this discussion to an immediate false statement.  We're a payment processor..  we're not a payment processor.  A small army of people with vested interests in this company rush to upvote each other's comments. A CEO says I'm talking down to them and proceeds to tell me their product is a general database.  I guess like MySQL?  I also said their DNS was MORE popular than Carnival Cruise- which he misread as less popular implying I was giving an insult- which is confusing.  I said RevenueCat DNS lookups get more traffic than (insert any big company) implying that being on this list has a miniscule effect in the world.  I hope that makes sense.

The only thing I'm looking at is the Revenue Cat website to see how the service is marketed.  The comments on here will not change my understanding of what I see is shown in product materials.

If this repo wants to whitelist it- that is their decision. If it becomes a thing that causes issues for AdAway I will address it there.  If AdAway users came to the AdAway list and said "something broke my Acme Co. subscription" it would've been a different situation.  Instead, RevenueCat tells me something is affecting their revenue.  It feels aggressive and concerning for RevenueCat to bring this up instead of someone who uses the list saying something is not working for them.

Regardless, your company can make these sorts of lists ineffective or unmaintainable.  DoubleClick, Swrve, optimizely, deltadna are a bunch of examples of companies that make host files impractical.  Your industry is all about finding ways to keep getting around these sort of hangups.

**Comment by jeiting** [comment]
Ah, re-reading your initial message I see now what you were saying. I must admit we feel unfairly maligned and probably are a bit defensive. Especially given we're very privacy conscious and are actively trying to help developers move away from the ads/data collection industry. 

Apologies for brigading you. This is affecting lots of our developers, who are receiving elevated customer service tickets, and so our default reaction is to jump to their defense. 

>  If AdAway users came to the AdAway list and said "something broke my Acme Co. subscription" it would've been a different situation.

I didn't go this route because I didn't want to flood you with inbound. Kind of happened anyway. Sorry.

Anyway, the world isn't on fire. Most end users of RevenueCat likely don't know what RevenueCat is though and won't make the connection that AdAway is breaking their ability to purchase. 

Have a good holiday and thanks.

**Comment by dnmTX** [comment]
I honestly....from my heart...**DO HOPE** that is the **END** of it !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by Cyral** [comment]
@jawz101 
> I walked into this discussion to an immediate false statement. We're a payment processor.. we're not a payment processor.

I understand your confusion, the entire purpose of RevenueCat is hard to understand if you aren't a mobile app developer who has dealt with in app purchases before.

Payment processors like Stripe have a beautiful API with webhooks when events happen (such as a customer upgrading, cancelling, etc). It is a joy to develop subscription services on their platform. On the other hand, Apple and Google have some of the worst payment APIs on the planet. Subscriptions weren't a thing when they were designed, so you have 10+ years of shoddy refactorings done to their payment services to "support" subscriptions. They are convoluted with lots of edge cases and completely idiotic design decisions. (see [step 6 in this article](https://medium.com/@AlexFaunt/auto-renewing-subscriptions-for-ios-apps-8d12b700a98f).

RevenueCat is a wrapper around these APIs that lets you write your payment code once, and have it work across both platforms using much more friendly APIs. They handle all of the crazy integration work (that would be too much for solo developers like myself to handle) on their end.

> If AdAway users came to the AdAway list and said "something broke my Acme Co. subscription" it would've been a different situation. 

I am an app developer who uses RevenueCat for the above reason and we are seeing issues because of this. AdWay users are likely not going to come to you about this issue, they are going to go to the app developer (or Apple/Google), if they even contact anyone at all (most users will just give up and move on). So I am coming to you on behalf of them to say, yes, it is breaking their subscriptions.

> A CEO says I'm talking down to them and proceeds to tell me their product is a general database. I guess like MySQL?

Many payment platforms allow you to include arbitrary data with payment profiles for quality of life purposes. Some companies want customer service to be able to see info (e.g. email or name) on a payment profile to make lookups either and reduce human error, although plenty of others roll their own (I have a dashboard to look up customers on _my_ end, and pull up their payment profile from there). It is purely a convenience thing, and in a way you are right, similar to MySQL - since apps are storing this data in their own database anyways if they collect it. (Yes, I am in a regulated industry and must collect user info)

I understand that your time is limited and that you must rely on the public facing websites to understand and make a decision on if a domain is ad/tracking related or not, I cannot expect you to be an expert in every industry. However I believe @rkotzy has accurately addressed some initial misconceptions you had from reading the website and I hope you will re-consider.

**Comment by jawz101** [comment]
I've been obsessed with this issue ever since it was mentioned.  And read and reread it all.  To understand my sensitivity, I work in an industry that deals with PII and when I see the data that goes into it.  When I see apps using [anonymous uids](https://docs.revenuecat.com/docs/user-ids) that sounds like something the app industry moved towards.  But then it's [this particular page](https://docs.revenuecat.com/docs/subscriber-attributes) with its particular screenshots and examples that I ams sensitive to.

`Purchases.shared.setAttributes(["age" : "24", 
                             "custom_group_id" : "abc123"])`



I think I understand what your primary business is.  I'm even listening to your podcast out of curiousity.  If this was just to manage subscriptions and act as a passthrough for other attribution and marketing platforms it would be fine.  If it breaks your company operational structure (e.g. your support portals) or the customer-to-company subscription service - which it sounds like it does break - I wouldn't want to block that.  But you have a component of your service with developer documentation where you give an example that age and hair color are the sorts of things that they can store on their end users in there accounts.  When I think of the principles of data minimization - [one of Google's own tenets](https://safety.google/intl/en_us/privacy/data/) - why does a magazine, workout, whatever subscription need my age and hair color?  Everything else about your product isn't these things and elsewhere you explain where your product fits into the [app monetization puzzle](https://www.revenuecat.com/blog/where-does-revenuecat-fit-in-your-app).

If you were a payment processor and this broke payment processing, that sounds important.  Your FAQs says you're not.  If your a middleman (subscription management) that sounds important but I don't know why you need my shoe size to manage a subscription.  That particular subscriber attribute feature and/or documentation examples bugs me.

As for this
`The subscriber attributes referenced in the PR to block api.revenuecat.com are an arbitrary key-value store, no different than Stripe metadata which does not appear to be blocked. Yes, some developers choose to use this metadata to collect things that would be "across your line" - just like with Stripe - and I can acknowledge that. But it's still not clear to me how that is different than any other database that you don't know how it's being used?`

Exactly.  They all scare the crap out of me.  I don't like that my personal attributes are sprinkled around on thousands of servers unaware with no way to opt-in vs opt-out, be notified what is there, or claw my data back.

In summary, scrap the shitty picture with my hair color and the age thing and I'd have never gone down this rabbit hole.  Maybe question if opt-in should be a required feature for this marginally insignificant user attributes function.  I mean come on.  If you knew your name, email, phone, _age, hair color_ was stored on a magazine subscription wouldn't you think that's odd?

**Comment by emcro** [comment]
I'm a one-man indie shop and found this thread trying to find the solution to more and more users reporting problems restoring purchases in my app. I integrated my in-app purchases + subscriptions with RevenueCat since late 2020 and it's made life *a ton* easier for me as a developer as I've been able to launch IAPs in just one day on iOS, and since integrated Stripe Checkout, and now Google Play. I do 0 creepy things with data, and only use that attribute system which you mention @jawz101 to know which screen in the app triggered the purchase flow.

The only way I can help these users at this point is to have them turn off PiHole, AdGuard, etc in order to restore a purchase. It truly sucks to be penalized in this way and to hurt my users' experience because of this decision. RevenueCat is literally the only third party piece of code I have in my entire app because I wanted to avoid exactly these kinds of issues, and their business model is simple and direct, I pay them a small % of the IAP revenue that they manage to make for a better user experience for all users -- buy once on any store, have access to all features across all platforms.

I really, really hope that RevenueCat can be removed from this blocklist, they're not one of the bad guys, and this decision is hurting a lot of indie developers like me.

**Comment by rkotzy** [comment]
I know a developer using RevenueCat that built an app for hairstylists to save color combinations for their clients. I can see `hair_color` being an attribute those developers may want to save 🤷‍♂️. Again, RevenueCat gives you a mechanism to store data, it doesn't collect data. And if it really comes down to that screenshot in the docs about Garfield's attributes, that can be changed. 

> If you were a payment processor and this broke payment processing, that sounds important. Your FAQs says you're not. If your a middleman (subscription management) that sounds important but I don't know why you need my shoe size to manage a subscription. That particular subscriber attribute feature and/or documentation examples bugs me.

You're 100% breaking the usability of apps that use RevenueCat because an api call to `api.revenuecat.com` is used to answer the question "Is this customer subscribed". Developers can't answer that question now from the app so paying customers lose access to content their entitled to. I don't think Stripe actually processed payments in the early days, they built clean APIs on top of legacy gateways and banks. 

> Maybe question if opt-in should be a required feature for this marginally insignificant user attributes function. I mean come on. If you knew your name, email, phone, age, hair color was stored on a magazine subscription wouldn't you think that's odd?

Would definitely be open for suggestions here. We thought making these opt-in for the developer would be sufficient. In the example of `hair_color`, the developer would need to ask the customer what their hair color was, the customer would need to provide that info, then the developer would store it in their database and could pass it to RevenueCat as well. Personally if I'm using an app and they start asking for creepy info I find unnecessary, I try to see if there's an option to not provide the info or delete the app. 

It's the background stuff that freaks me out too, which is why I use PiHole.

**Comment by jawz101** [comment]
@rkotzy well I'm removing it from the blocklist.  Clearly this is a feature that has its place but it might at least be nice to make some of those particular use cases within the documentation actually walk through a particular and less creepy example.  That is, instead of just making up "here's where you can store their name, email, age, number, hair color." Maybe use a video store and their taste in horror films.  Having so many demographics in the examples scares the shit out of me and it conditions some developers to go ahead and follow your documentation word-for-word and add all of these fields as if they are your standard app fields.  It just doesn't encourage data minimization to use arbitrary examples.

I don't know what liabilities come with storing data as a hosting provider for someone else's company but I'd encourage at least having a sensible example for a particular, bland product instead of personal info.

I mean [here's an example ](https://www.nbcnews.com/tech/tech-news/amazon-accused-violating-children-s-privacy-kid-friendly-smart-speakers-n1003706) of a concern where Amazon got attention for 3rd party Amazon Skills on Echo Dots.  You may want to consider having a less arbitrary example and saying something about data minimization principles somewhere.  I just don't like the example there.


Sorry I'm being picky but that's where my head goes.  Anyways, I removed the entry but hopefully I made a little sense.

**Comment by jeiting** [comment]
I think we can all agree the hair color thing is creepy. There are better, more appropriate examples we should use since it will suggest to our users how to use the feature. Sorry, it was just a bad "foo" value.

**Comment by emcro** [comment]
Thank you so much for working through this, restoring RevenueCat on PiHole and other services is going to make our support lives *a lot* easier. Thank goodness I don't have to deal with all the app receipt mess myself across 3 platforms, dev life is hard enough as it is. Really appreciate it, hope you have a great new year!
